lgc/builder/BuilderImpl.cpp

```diff
@@ -117,8 +119,15 @@ Value *BuilderImplBase::CreateIntegerDotProduct(Value *vector1, Value *vector2,
       Value *product = CreateMul(elem1, elem2);
       scalar = CreateAdd(product, scalar);
```

Comment:
Does clamping work correctly for i32 element input and i32 result? I think it is possible for this add to overflow and wrap, such that the clamp to 32 bits at the end does not clamp. Suppose we do a dot2 on <0xffffffff,0x10000> with <0xffffffff,0x20000>. The two values of "scalar" are 0xfffffffe00000001 and 0x0000000200000000, and the add here wraps to 0x00000000000000001. Or have I got that wrong?

Reply:
@trenouf If there is an accumulator, the spv spec said "If any of the multiplications or additions, with the exception of the final accumulation, overflow or underflow, the result of the instruction is undefined." So for your case, the result is should be undefined and we don't need special handling, is it right?
I add the comments for the statements of the result in the spec.